command/agent/auth/auth.go

```diff
@@ -190,6 +198,11 @@ func (ah *AuthHandler) Run(ctx context.Context, am AuthMethod) error {
 			secret, err = clientToUse.Auth().Token().LookupSelfWithContext(ctx)
 			if err != nil {
 				ah.logger.Error("could not look up token", "err", err, "backoff", backoff)
+
+				if ah.exitOnError {
+					return err
+				}
+
 				backoffOrQuit(ctx, backoff)
```

Comment:
Not 100% related to your PR, but it looks like `backoffOrQuit` won't ever actually quit. I wonder if it would make some sense to try to simplify the exit logic somehow with `ah.exitOnError` to make `backoffOrQuit` actually quit (and do so without backing off if `ah.exitOnError`?

Reply:
I reworked it a bit. Definitely feels better now.

Reply:
:+1: